libraries/ui/src/main/java/androidx/media3/ui/SubtitlePainter.java

```diff
@@ -470,6 +480,62 @@ private void drawTextLayout(Canvas canvas) {
     canvas.restoreToCount(saveCount);
   }
 
+  private void drawPadding(Canvas canvas, Paint paint, int color) {
+    if (horizontalPadding == 0) {
+      return;
+    }
+    // save original color
+    int paintColor = paint.getColor();
+    int bgColor = color;
+    if (textLayout.getText() instanceof Spannable) { // Spannable text can set background.
+      BackgroundColorSpan[] colorSpans = ((Spannable) textLayout.getText())
+          .getSpans(0, textLayout.getText().length(), BackgroundColorSpan.class);
+      if (colorSpans != null && colorSpans.length > 0) {
+        bgColor = colorSpans[colorSpans.length - 1].getBackgroundColor();
+      }
+    }
+    // set background color
+    paint.setColor(bgColor);
+
+    for (int line = 0; line < textLayout.getLineCount(); line++) {
+      backgroundPaddingRect.top = textLayout.getLineTop(line);
+      backgroundPaddingRect.bottom = textLayout.getLineBottom(line);
+
+      if (Color.alpha(bgColor) == 0xFF) { // draw all the regions
+        backgroundPaddingRect.left = textLayout.getLineLeft(line) - horizontalPadding;
+        backgroundPaddingRect.right = textLayout.getLineRight(line) + horizontalPadding;
+        canvas.drawRect(backgroundPaddingRect, paint);
+      } else {
+        float left = textLayout.getLineLeft(line);
+        float right = textLayout.getLineRight(line);
+        if (textLayout.getAlignment() == Alignment.ALIGN_CENTER) {
```

Comment:
Why do we need to handle this separately? Because of multi lines?

Reply:
here is textLayout cal x position logic:
![image](https://github.com/user-attachments/assets/77937d42-f967-42ce-9f17-53f6905ae1a5)

getLineLeft and getLineRight is slightly different than the exact result in center mode.
getLineLeft will return 90.5, but actually is 91.